pkg/tsdb/cloudwatch/routes/middleware.go

```diff
@@ -0,0 +1,16 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+)
+
+func RouteInjector(handleFunc models.RouteHandlerFunc, clientFactory models.ClientsFactoryFunc) func(rw http.ResponseWriter, req *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		pluginContext := httpadapter.PluginConfigFromContext(ctx)
+		handleFunc(rw, req, clientFactory, pluginContext)
+	}
+}
```

Comment:
Not sure why this would be better? 

Even though the middleware is pretty sparse right now, I think as we add more routes we're gonna find that all of them are handling GET requests. Then each handler func could return `[]bytes (json), int, (statusCode), error`. That should simplify the routes a lot. Also, it will allow us to provide a generic approach to logging errors, which is currently not done at all today.

Reply:
@fridgepoet pushed a [new commit](https://github.com/grafana/grafana/pull/57148/commits/0202ddd920c5b2317ac29b860461595e82905eea#diff-87ed98b4620649dd11ec08ac66c594520315e4c7fbda4eaf132dfc3df3ed3cd6) that I think highlights the benefits of using a middleware for the repetitive parts. You'll have to imagine that we in the future we'll have around 10 handlers.

Reply:
Ok thanks for that!